### ISSUE TITLE
Rust syntax is only checked on save

### ISSUE THREAD

**Issue by Visne** [flycheck/flycheck#1954]
## Checklist
- [x] I have checked existing issues for potential duplicates before creating this one.
- [x] I have read the [Troubleshooting guide][].

## Bug description
Rust syntax is only checked when the file is saved. This means that errors that are already fixed, are still marked in red. Also, this means that code actions can be applied twice, which usually ruins the code instead of fixing it.

## Steps to reproduce
1. Open a Rust file with correct syntax.
1. Make the syntax incorrect by typing random letters into a function call.
1. No error is shown by Flycheck.
1. Save the file.
1. An error is shown by Flycheck.
1. Fix the error.
1. The error is still shown by Flycheck.
1. Save the file.
1. No error is shown by Flycheck.

- [x] I have read https://emacs.stackexchange.com/questions/28429/how-do-i-troubleshoot-emacs-problems

(tell me if I have to provide a bug recipe, I think it is redundant for this issue)

## Expected behavior
1. Open a Rust file with correct syntax.
1. Make the syntax incorrect by typing random letters into a function call.
1. An error is shown by Flycheck.
1. Fix the error.
1. No error is shown by Flycheck.

## System configuration
```
Syntax checkers for buffer main.rs in rustic-mode:

First checker to run:

  lsp (explicitly selected)
    - may enable: yes
    - may run:    t

Checkers that could run if selected:

  rustic-clippy  select
    - may enable: yes
    - may run:    t
    - executable: Found at /usr/bin/cargo
    - Clippy:     Cannot find the `cargo clippy' command
    - Cargo.toml: Found

Flycheck Mode is enabled.  Use C-u C-c ! x to enable disabled
checkers.

--------------------

Flycheck version: 33snapshot
Emacs version:    28.1
System:           x86_64-pc-linux-gnu
Window system:    x
```

Emacs configuration:
- [x] Plain Emacs / Custom configuration
- [ ] Spacemacs
- [ ] Doom Emacs
- [ ] Other shared configuration

[Troubleshooting guide]:  https://www.flycheck.org/en/latest/user/troubleshooting.html

**Comment by bertulli** [comment]
I have the same problem. I can attach some screenshot for clarification:
Normal file:
![Screenshot from 2022-08-01 12-28-15](https://user-images.githubusercontent.com/10565357/182132121-26b55803-eeb2-4a65-8ce1-22fac07b6041.png)

After having modified the buffer:
![Screenshot from 2022-08-01 12-28-45](https://user-images.githubusercontent.com/10565357/182132166-15e18568-21f0-42c4-8e48-b8de5d97527b.png)

After save (`C-x C-s`):
![Screenshot from 2022-08-01 12-35-25](https://user-images.githubusercontent.com/10565357/182132236-8e901ee8-fb81-4945-a67a-ea576b5b15f5.png)

**Comment by chubbymomo** [comment]
Do you have any updates on this issue? I can confirm that I also have this issue when running with flycheck + lsp-mode and rustic.

**Comment by jcs090218** [comment]
Hmmm... What's the value of `flycheck-check-syntax-automatically`? Does this happen when flymake + lsp-mode? 🤔

**Comment by bertulli** [comment]
> Hmmm... What's the value of `flycheck-check-syntax-automatically`?

It is `'(save idle-change new-line mode-enabled)`. However, apparently it seems that this is actually expected behaviour, especially [on `rust-analyzer`'s side](https://users.rust-lang.org/t/rust-analyzer-doesnt-check-the-buffer-on-typing-but-only-on-save-how-to-change-that/79221/12). So it's not a flycheck issue, I think (for instance with `clangd` it semmes to work for me), therefore if you want to close this issue it's ok for me.

Thanks!
